File: java/platform-data-doublets-client/src/test/java/platform/data/doublets/client/LinkTest.java
package platform.data.doublets.client;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class LinkTest {

    @Test
    void testLinkCreation() {
        Link link = new Link(1L, 2L, 3L);
        
        assertEquals(1L, link.getId());
        assertEquals(2L, link.getFromId());
        assertEquals(3L, link.getToId());
    }

    @Test
    void testLinkEquality() {
        Link link1 = new Link(1L, 2L, 3L);
        Link link2 = new Link(1L, 2L, 3L);
        Link link3 = new Link(2L, 2L, 3L);

        assertEquals(link1, link2);
        assertNotEquals(link1, link3);
        assertEquals(link1.hashCode(), link2.hashCode());
    }

    @Test
    void testLinkToString() {
        Link link = new Link(1L, 2L, 3L);
        String expected = "Link{id=1, fromId=2, toId=3}";
        
        assertEquals(expected, link.toString());
    }
}

File: php/tests/LinkTest.php
<?php

declare(strict_types=1);

namespace LinksPlatform\Data\Doublets\Gql\Tests;

use LinksPlatform\Data\Doublets\Gql\Link;
use PHPUnit\Framework\TestCase;

class LinkTest extends TestCase
{
    public function testLinkCreation(): void
    {
        $link = new Link(1, 2, 3);
        
        $this->assertEquals(1, $link->getId());
        $this->assertEquals(2, $link->getFromId());
        $this->assertEquals(3, $link->getToId());
    }

    public function testFromArray(): void
    {
        $data = [
            'id' => 10,
            'from_id' => 20,
            'to_id' => 30
        ];
        
        $link = Link::fromArray($data);
        
        $this->assertEquals(10, $link->getId());
        $this->assertEquals(20, $link->getFromId());
        $this->assertEquals(30, $link->getToId());
    }

    public function testToArray(): void
    {
        $link = new Link(1, 2, 3);
        $expected = [
            'id' => 1,
            'from_id' => 2,
            'to_id' => 3
        ];
        
        $this->assertEquals($expected, $link->toArray());
    }

    public function testToJson(): void
    {
        $link = new Link(1, 2, 3);
        $expected = '{"id":1,"from_id":2,"to_id":3}';
        
        $this->assertEquals($expected, $link->toJson());
    }

    public function testToString(): void
    {
        $link = new Link(1, 2, 3);
        $expected = 'Link(id: 1, from: 2, to: 3)';
        
        $this->assertEquals($expected, (string)$link);
    }
}